packages/core/test/result-transformers.test.ts

```diff
@@ -268,6 +268,153 @@ describe('resultTransformers', () => {
+          new ResultSummary<Integer>(query, params, meta)
+        )
+
+        expect(typeAssertionBigInt).toEqual(
+          new ResultSummary<Integer>(query, params, meta)
+        )
+      })
+
+      it('it should return a ResultSummary<bigint>', async () => {
+        const resultStreamObserverMock = new ResultStreamObserverMock()
+        const query = 'Query'
+        const params = { a: 1 }
+        const meta = { db: 'adb' }
+        const result = new Result(Promise.resolve(resultStreamObserverMock), query, params)
+        const keys = ['model', 'year']
+        const rawRecord1 = ['Beautiful Sedan', 1987]
+        const rawRecord2 = ['Hot Hatch', 1995]
+
+        resultStreamObserverMock.onKeys(keys)
+        resultStreamObserverMock.onNext(rawRecord1)
+        resultStreamObserverMock.onNext(rawRecord2)
+        resultStreamObserverMock.onCompleted(meta)
+        const summary = await resultTransformers.summary<bigint>()(result)
+
+        // @ts-expect-error
+        const typeAssertionNumber: ResultSummary<number> = summary
+        // @ts-expect-error
+        const typeAssertionInteger: ResultSummary<Integer> = summary
+
+        expect(typeAssertionNumber).toEqual(
+          new ResultSummary<Integer>(query, params, meta)
+        )
+
+        expect(typeAssertionInteger).toEqual(
+          new ResultSummary<Integer>(query, params, meta)
+        )
+      })
+
+      it('it should return a ResultSummary<Integer>', async () => {
+        const resultStreamObserverMock = new ResultStreamObserverMock()
+        const query = 'Query'
+        const params = { a: 1 }
+        const meta = { db: 'adb' }
+        const result = new Result(Promise.resolve(resultStreamObserverMock), query, params)
+        const keys = ['model', 'year']
+        const rawRecord1 = ['Beautiful Sedan', 1987]
+        const rawRecord2 = ['Hot Hatch', 1995]
+
+        resultStreamObserverMock.onKeys(keys)
+        resultStreamObserverMock.onNext(rawRecord1)
+        resultStreamObserverMock.onNext(rawRecord2)
+        resultStreamObserverMock.onCompleted(meta)
+        const summary = await resultTransformers.summary<Integer>()(result)
+
+        // @ts-expect-error
+        const typeAssertionNumber: ResultSummary<number> = summary
+        // @ts-expect-error
+        const typeAssertionBigInt: ResultSummary<bigint> = summary
+
+        expect(typeAssertionNumber).toEqual(
+          new ResultSummary<Integer>(query, params, meta)
+        )
+
+        expect(typeAssertionBigInt).toEqual(
+          new ResultSummary<Integer>(query, params, meta)
+        )
+      })
+    })
+
+    describe('when results fail', () => {
+      it('should propagate the exception', async () => {
+        const expectedError = newError('expected error')
+        const result = new Result(Promise.reject(expectedError), 'query')
+
+        await expect(resultTransformers.eagerResultTransformer()(result)).rejects.toThrow(expectedError)
```

Comment:
```suggestion
        await expect(resultTransformers.summary()(result)).rejects.toThrow(expectedError)
```
Testing the wrong transformer?

Reply:
copy paste error